### PR TITLE
Redstone triggered script blocks can now be included in structures an…

### LINKED ISSUE #269: [1.10.2-1.4.5] Script blocks redstone activation not usable in structures
If a Script block is allowed to trigger on spawn, it works fine, but if set to only trigger on redstone, it still removes itself instantly when the structure it's in generates, so it never gets a chance to be activated. 

As such, my plans for spawning a substructure under certain conditions cannot work, since these script blocks are (as far as I can tell) the only way to spawn structures automatically other than worldgen, since the /#gen command requires the /#confirm followup so cannot be used to gen structures in regular command blocks.

EDIT: Okay, I stand corrected, /#gen works in command blocks, but I can't set the relative y level that way, it just uses the placement from the structure it's genning, which voids its usefulness when I'm trying to gen a basement under a building

### PR DESCRIPTION
**Pull request by Mordenkainen**
…d will remain until triggered.

Fixes #269

### CODE REVIEW

File: src/main/java/ivorius/reccomplex/block/TileEntityBlockScript.java [hunk 1]
```diff
@@ -68,13 +68,14 @@ public NBTTagCompound writeToNBT(NBTTagCompound nbtTagCompound)
...
-        script.generate(context, instanceData, pos);
+    	if (spawnTriggerable || (redstoneTriggerable && redstoneTriggered))
+    		script.generate(context, instanceData, pos);
     }
 
     @Override
     public boolean shouldPlaceInWorld(StructureSpawnContext context, WorldScriptMulti.InstanceData instanceData)
     {
-        return false;
+        return !spawnTriggerable || (redstoneTriggerable && !redstoneTriggered);
```
Comment (Ivorforce) [inline src/main/java/ivorius/reccomplex/block/TileEntityBlockScript.java]: Why should scripts that trigger on spawn _and_ redstone input, but are currently powered not spawn?

File: src/main/java/ivorius/reccomplex/block/TileEntityBlockScript.java [hunk 2]
```diff
@@ -68,13 +68,14 @@ public NBTTagCompound writeToNBT(NBTTagCompound nbtTagCompound)
     @Override
     public void generate(StructureSpawnContext context, WorldScriptMulti.InstanceData instanceData)
     {
-        script.generate(context, instanceData, pos);
+    	if (spawnTriggerable || (redstoneTriggerable && redstoneTriggered))
```
Comment (Ivorforce) [inline src/main/java/ivorius/reccomplex/block/TileEntityBlockScript.java]: Why do you start testing for spawnTriggerable (or redstoneTriggerable too) here? The spawnTriggerable check in prepareInstanceData takes care of determining when it should spawn.

File: src/main/java/ivorius/reccomplex/block/TileEntityBlockScript.java (hunk 1 above)
Comment (Mordenkainen) [reply src/main/java/ivorius/reccomplex/block/TileEntityBlockScript.java]: Sorry, now I get it, it's and AND not OR. The latest commit should properly handle cases where the script block is set to spawn AND redstone triggers.

**Comment by Ivorforce** [comment]
Thanks :) I ended up making my own commit though because of the other changes you had made in unrelated files.

**Comment by Mordenkainen** [comment]
Sorry about that. I was in a bit of a rush.

**Pull request closed without merging**